File: membrane/cytoskeleton/.archived-pre-consolidation/interoceptor.js
#!/usr/bin/env node
/**
 * Notification Hook - Log background task completions
 *
 * Logs notifications to a file so background task results
 * aren't silently lost if the agent misses them.
 */

const fs = require('fs');
const path = require('path');

const LOG_FILE = path.join(process.env.HOME, 'logs', 'notification-log.jsonl');

let input = '';
process.stdin.on('data', chunk => input += chunk);
process.stdin.on('end', () => {
  try {
    const data = JSON.parse(input);
    const now = new Date().toISOString();

    const entry = JSON.stringify({
      timestamp: now,
      type: data.type || 'unknown',
      message: data.message || '',
      tool: data.tool_name || '',
    });

    // Ensure log directory exists
    const logDir = path.dirname(LOG_FILE);
    if (!fs.existsSync(logDir)) {
      fs.mkdirSync(logDir, { recursive: true });
    }

    fs.appendFileSync(LOG_FILE, entry + '\n');
  } catch {
    // Don't block on errors
  }
  process.exit(0);
});

File: membrane/cytoskeleton/interoceptor.py
#!/usr/bin/env python3
"""interoceptor.py — Notification hook. Logs background task completions."""

from __future__ import annotations

import json
import sys
from datetime import datetime
from pathlib import Path

LOG_FILE = Path.home() / "logs" / "notification-log.jsonl"

try:
    data = json.load(sys.stdin)
    entry = json.dumps(
        {
            "timestamp": datetime.now().isoformat(timespec="seconds"),
            "type": data.get("type", "unknown"),
            "message": data.get("message", ""),
            "tool": data.get("tool_name", ""),
        }
    )
    LOG_FILE.parent.mkdir(parents=True, exist_ok=True)
    with LOG_FILE.open("a") as f:
        f.write(entry + "\n")
except Exception:
    pass
